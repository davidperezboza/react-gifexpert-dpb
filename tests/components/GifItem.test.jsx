const {render, screen} = require('@testing-library/react');
const { GifItem } = require('../../src/components/GifItem');

describe('Prueba de GifItem', () => { 
    const title = 'Titulo de prueba';
    const url = 'http://urldeprueba.com/';

    test('Debe de hacer match con el snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot(); 
    }); 

    test('debe de mostrar la imagen con el url y el alt indicado', () => {
        render(<GifItem title={title} url={url}/>);
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe de mostrar el titulo en componente', () => { 
        render(<GifItem title={title} url={url}/>); 
        expect(screen.getByText(title)).toBeTruthy();
    });

});