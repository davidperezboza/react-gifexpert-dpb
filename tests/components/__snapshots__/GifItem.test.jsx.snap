// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo de prueba"
      src="http://urldeprueba.com/"
    />
    <p>
      Titulo de prueba
    </p>
  </div>
</div>
`;
